---
title: "bcdataset"
author: "Lei Jiang"
date: "March 17, 2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown



```{r}
bc<-read.table("https://archive.ics.uci.edu/ml/machine-learning-databases/breast-cancer-wisconsin/wdbc.data",header=F,sep=",")
names(bc)<- c('id_number', 'diagnosis', 'radius_mean', 
              'texture_mean', 'perimeter_mean', 'area_mean', 
              'smoothness_mean', 'compactness_mean', 
              'concavity_mean','concave_points_mean', 
              'symmetry_mean', 'fractal_dimension_mean',
              'radius_se', 'texture_se', 'perimeter_se', 
              'area_se', 'smoothness_se', 'compactness_se', 
              'concavity_se', 'concave_points_se', 
              'symmetry_se', 'fractal_dimension_se', 
              'radius_worst', 'texture_worst', 
              'perimeter_worst', 'area_worst', 
              'smoothness_worst', 'compactness_worst', 
              'concavity_worst', 'concave_points_worst', 
              'symmetry_worst', 'fractal_dimension_worst')

#write.csv(bc,file="C:/Users/N1110/Desktop/bcdataset.csv")
```

## Including 

```{r, echo=FALSE}
dim(bc)
str(bc)
head(bc)

```

#subset  
```{r}
bc2<-bc[,c(1,2)]
bc2
```
#simulation data 569 samples, 21 genes genomic test data. generate a 569x21 matrix for patients genomic test dataset with gene expression levels range from 1-3. (These are all cancer patients data, so we do not have 0 value since it means no cancer. we do not know the actual underlie distribution though. so just used a normal distribution)
# ref: Carlsson et al. paper in 2004. HER2 expression in breast cancer primary tumours and corresponding metastases. Original data and literature review
#HER2-scores: The HER2 expression was scored using the HercepTest criteria. The HER2-score was based on a 0 to 3+ scale. 0 corresponded to tumour cells that were completely negative, 1+ corresponded to faint perceptible staining of the tumour cell membranes, 2+ corresponded to moderate staining of the entire tumour cell membranes and 3+ indicated strong circumferential staining of the entire tumour cell membranes creating a fishnet pattern. 
```{r}

genomat=matrix(sample.int(3, 569*21, TRUE),569,21) 

genomat2=data.frame(genomat)

head(genomat2)
```

#
```{r}
names(genomat2)<- c('Grb7', 'HER2', 'ER','PR', 'BCL2', 'SCUBE2', 'Ki67', 'STK15', 'Survivin', 'CyclinBI', 'MYBL2', 'MMPII', 'CTSL2', 'CD68', 'GSTMI', 'BAG1', 'Bactin', 'GAPDH', 'RPLPO', 'GUS', 'TFRC')
head(genomat2)
```
#bind patients ID and diagnosis data with genotyping data to form a complete table
```{r}
GenoType=cbind(bc2, genomat2)
head(GenoType)
```

#write to csv
```{r}
write.csv(GenoType,file="C:/Users/N1110/Desktop/GenoType.csv")
```

